sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -905,8 +1054,19 @@ public ServiceBusSenderClientBuilder topicName(String topicName) {
          *     {@link #queueName(String) queueName} or {@link #topicName(String) topicName}.
          * @throws IllegalArgumentException if the entity type is not a queue or a topic.
          */
+        // Build Sender-Client.
         public ServiceBusSenderAsyncClient buildAsyncClient() {
-            final ServiceBusConnectionProcessor connectionProcessor = getOrCreateConnectionProcessor(messageSerializer);
+            final ConnectionCacheWrapper connectionCacheWrapper;
+            final Runnable onClientClose;
+            final boolean isSenderOnV2 = !v2StackSupport.isSenderAndManageRulesApiOptedOut(configuration);
```

Comment:
It was something I was thinking - _OptOut|OptIn_ vs. _Enabled|Disabled_. In fact, it was _Enabled_ initially; then I felt its readable the other way, as its user opting here, while _Enable_ may get confused with the actual enable of feature (for example, is RECEIVE is enabled for the queue in the Portal), didn't think much as this a temporary type anyway. Did you find Opt*** usage confusing?

Reply:
no strong preference. I just noticed we use this API with double negation (`!v2StackSupport.isSenderAndManageRulesApiOptedOut`), so it might be more straightforward to have `is*Enabled` API. In any case, it's not public and we can always refactor it

Reply:
Sounds good; regarding the negation, it was intentional, i.e., flag is enabled by default, so if it is not explicitly opted out, we will host Sender|Mgmt on v2.

Reply:
the default value is definitely important when picking public configuration property name, but names of our internal APIs can be reverse. In any case, I fully trust your choice :)

Reply:
Let me think again about the name as part of using the 'ConfigurationProperty' that you suggested and get back, thanks you!

Reply:
As part of incorporating 'ConfigurationProperty', I've improved the readability by using the suffix "**Enabled".